app/Providers/AuthServiceProvider.php

```diff
@@ -23,5 +27,13 @@ class AuthServiceProvider extends ServiceProvider
 	public function boot(): void
 	{
 		$this->registerPolicies();
+
+		Gate::define('admin', function (User $user) {
+			return $user->id === 0;
+		});
+
+		Gate::define('can-upload', function (?User $user) {
+			return optional($user)->may_upload === true;
+		});
```

Comment:
As said in some earlier comment on `AlbumPolicy` aka `AlbumAuthorisationProvider` and `PhotoPolicy` aka `PhotoAuthorisationProvider`, these gates are mostly used there (not entirely). Moreover, they are so trivial that they introduce just another layer of indirection which does not do anything. Moreover, we already have `User:isAdmin` which does the exact some comparison `=== 0`. We should not have this twice.

Reply:
fixed.
Remove the isAdmin() method from User and moved it to Policy.
This will allow use later to support multi admin and more granularity in user rights.